src/microstates.js

```diff
@@ -1,28 +1,30 @@
+import { map } from 'funcadelic';
 import Tree from './utils/tree';
 import States from './utils/states';
 import Transitions from './utils/transitions';
 import validate from './utils/validate';
 
-export default class Microstates {
-  constructor(tree, value) {
-    this.states = States(tree, value).collapsed;
-    this.transitions = Transitions(tree, this.states).collapsed;
-    this.tree = tree;
-  }
-  /**
-   * Create new Microstates for same type tree but new value.
-   * @param value any
-   */
-  to(value) {
-    return new Microstates(this.tree, value);
-  }
-  /**
-   * Build Microstates for given type and value.
-   * @param Type Tree
-   * @param value any
-   */
-  static from(Type, value) {
-    validate(Type, `microstates`);
-    return new Microstates(Tree.from(Type), value);
-  }
+const { assign } = Object;
+
+export default function Microstates(Type) {
+  let tree = Tree.from(Type);
+
+  return function Microstate(value) {
+    let states = States(tree, value).collapsed;
+    let transitions = Transitions(tree, states, value);
+
+    return assign(
```

Comment:
Curious why `assign` here and not `append`. This will eagerly evaluate the object that you're mapping with line 26

Reply:
They would both work. In this case, it seems that `assign` was the least powerful tool for the job.

Reply:
I originally used object spread but it's not available before Node 8.